pulp/apis/highs_api.py

```diff
@@ -0,0 +1,163 @@
+            warnings.warn('HiGHS_CMD does not currently allow maximization, '
+                          'we will minimize the inverse of the objective function.')
+            lp += -lp.objective
+        lp.checkDuplicateVars()
+        lp.checkLengthVars(52)
+        lp.writeMPS(tmpMps, mpsSense=constants.LpMinimize)
+
+        # just to report duplicated variables:
+        try:
+            os.remove(tmpSol)
+        except:
+            pass
+        cmd = self.path
+        cmd += ' %s' % tmpMps
+        cmd += ' --options_file %s' % tmpOptions
+        if self.timeLimit is not None:
+            cmd += ' --time_limit %s' % self.timeLimit
+        for option in self.options:
+            cmd += ' ' + option
+        if lp.isMIP():
+            if not self.mip:
+                warnings.warn("HiGHS_CMD cannot solve the relaxation of a problem")
+        if self.msg:
+            pipe = None
+        else:
+            pipe = open(os.devnull, 'w')
+        lp_status = None
+        with subprocess.Popen(cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True) as proc, open(tmpLog, "w") as log_file:
+            for line in proc.stdout:
+                if self.msg: sys.__stdout__.write(line)
+                log_file.write(line)
+
+        # We need to undo the objective swap before finishing
+        if lp.sense == constants.LpMaximize:
+            lp += -lp.objective
+        
+        # The return code for HiGHS follows: 0:optimal, 1: Iteration/time limit, 2: Infeasible, 3: Unbounded, 4: Solver error. See the return status here: https://docs.scipy.org/doc/scipy/reference/optimize.linprog-highs.html
+        if proc.wait() == 0:
+            status = constants.LpStatusOptimal
+            status_sol = constants.LpSolutionOptimal
+        elif proc.wait() == 1:
+            status = constants.LpStatusOptimal # this is following the PuLP convention; in case, there is no feasible solution, the empty solution file will update the status of problem and solution later
+            status_sol = constants.LpSolutionIntegerFeasible
+        elif proc.wait() == 2:
+            status = constants.LpStatusInfeasible
+            status_sol = constants.LpSolutionNoSolutionFound
+        elif proc.wait() == 3:
+            status = constants.LpStatusUnbounded
+            status_sol = constants.LpSolutionNoSolutionFound
+        elif proc.wait() == 4:
+            status = constants.LpStatusNotSolved
+            status_sol = constants.LpSolutionNoSolutionFound
+        else:
+            status = constants.LpStatusUndefined
+            status_sol = constants.LpSolutionNoSolutionFound
+            raise PulpSolverError("Pulp: Error while executing", self.path)
+            
+        if not os.path.exists(tmpSol):
+            status = constants.LpStatusNotSolved
+            status_sol = constants.LpSolutionNoSolutionFound
+        
+        values, status, status_sol = self.readsol(tmpSol, status, status_sol)
```

Comment:
instead of putting the status check inside the `readsol` I think it's simpler to match what we do in the mipcl api:

```python
        if not os.path.exists(tmpSol):
            status = constants.LpStatusNotSolved
            status_sol = constants.LpSolutionNoSolutionFound
            values = None
        else:
            values = self.readsol(tmpSol)
```
and then your `readsol` assumes the filename exists. Also this way you won't need the `status` or `status_sol` arguments and you won't need to return them.